Source/BoundaryConditions/PML.H

```diff
@@ -112,7 +112,7 @@ public:
     }
 
 private:
-    const amrex::BoxArray& m_grids;
+    amrex::BoxArray m_grids;
```

Comment:
Why not simply a ref?
```suggestion
    amrex::BoxArray & m_grids;
```

Reply:
@ax3l , the goal of [avoid-const-or-ref-data-members](
https://releases.llvm.org/16.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/cppcoreguidelines/avoid-const-or-ref-data-members.html) is to avoid using `const` or `ref` data members, on the grounds that
>  Having such members is rarely useful, and makes the class only copy-constructible but not copy-assignable

This recommendation is part of the [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c12-dont-make-data-members-const-or-references), and `clant-tidy` documentation suggests also this [additional reading](https://quuxplusone.github.io/blog/2022/01/23/dont-const-all-the-things/#data-members-never-const) on the topic.

That said, we are not forced to enforce this rule in WarpX (although I think that this rule makes sense in most cases), or we could make an exception specifically for this line. For the moment I've implemented @WeiqunZhang 's suggestion, but I can revert it and just add a `NOLINT`.

Reply:
Agreed, but my suggestion just adds `&` not `const` back :D